### PR TITLE
Picker updates

### PR DESCRIPTION
**Pull request by jenwilhelm**
Adds several new features to the picker:
- Search by shortcode ([issue 22](https://github.com/wedgies/jquery-emoji-picker/issues/22))
- Recently Used emojis section (new feature)
- Scroll view instead of tabs ([issue 27](https://github.com/wedgies/jquery-emoji-picker/issues/27))
- New category icons
- Emoji hover effect + info view (new feature)
- Updated styling (increased gray contrast, borders on tabs)
- Adds new Unicode 8 emojis ([issue 31](https://github.com/wedgies/jquery-emoji-picker/issues/31))
  emojipicker

The design and functionality is based heavily on the emoji picker in Slack. :balloon:

### CODE REVIEW

**Comment by jenwilhelm** [comment]
@makenai @zpnk This is ready for review! :tada: I believe everything is working correctly except for emojis with a range... the family + couple emojis under People are not showing up, yet. I'm commenting these out of the master json file until I figure it out.

**Comment by makenai** [comment]
I have a small stylistic request / suggestion. When the emoji picker first opens and nobody is being moused over, the emoji detail area is blank and looks a bit funny.

screen shot 2016-01-07 at 6 13 53 pm

What if we:

1) Started with it hidden and then slid it up when someone hovered

or 

2) Wrote something fun there like an emoji tip of the day or an inspirational message so it doesn't look so blank

**Comment by makenai** [comment]
Similarly, what if when we are hovering over the categories, it would show the category icon in the detail area and describe what it was 'Recently Used', 'Plants', 'Food' etc. Maybe also list the number of emoji in that tab.

That might be overkill, but adding a title="" with the description of the category would be helpful too.

screen shot 2016-01-07 at 6 18 45 pm

File: js/jquery.emojipicker.js [hunk 1]
```diff
@@ -124,15 +126,27 @@
       }
 
       // Click event for emoji
-      this.$picker.find('section div')
-        .click( $.proxy(this.emojiClicked, this) );
+      this.$picker.on('click', 'em', $.proxy(this.emojiClicked, this));
+
+      // Hover event for emoji
+      this.$picker.on('mouseover', 'em', $.proxy(this.emojiMouseover, this) );
+      this.$picker.on('mouseout',  'em', $.proxy(this.emojiMouseout, this) );
```
Comment (makenai) [inline js/jquery.emojipicker.js:144]: Yay, so many fewer events!

**Comment by makenai** [comment]
I think there is a bug when I start in incognito mode and empty 'Recently Used' it never populates and causes a JS error. I think it happened to me when I first used it too, rather than being limited to a localstorage, but it's worth looking in to.

localstoragecleared

Clicking on the recently used tab creates the following error:

screen shot 2016-01-07 at 6 27 27 pm

**Comment by makenai** [comment]
Great work, Jen! the new updates are awesome and make the emoji picker function and look WORLDS better. :heart_eyes: :heart_eyes: :heart_eyes:

**Comment by jenwilhelm** [comment]
@makenai Damn you, Pawel, and your great suggestions. :beers: I just pushed some additional changes! :tropical_drink: :grin:

**Comment by makenai** [comment]
Fixes look great! I'm going to merge it.

I have two last minor nitpicks:

1) We should call the Daily Emoji the Random Emoji, since it doesn't stick around for the entire day
2) I think it should be selectable!

I think I can do a PR back at you for that though.

**Pull request merged**